Adding new rule for RHEL-08-010294

carlosmmatos opened the pull request:
#### Description:

- Adding new rule for RHEL 8 STIG - RHEL-08-010294
- Will make use of #6953 

#### Rationale:

- The RHEL 8 operating system must implement DoD-approved TLS encryption in the OpenSSL package.

carlosmmatos commented:
/retest

ggbecker commented on linux_os/guide/system/software/integrity/crypto/configure_openssl_tls_crypto_policy/ansible/shared.yml:
```diff
@@ -0,0 +1,16 @@
+# complexity = low
+# disruption = low
+
+- name: "Ensure system crypto policy has OpenSSL library using TLS encryption"
+  lineinfile:
+    path: "/etc/crypto-policies/back-ends/opensslcnf.config"
+    line: "MinProtocol = TLSv1.2"
+    create: yes
+    insertafter: EOF
```
```suggestion
{{{ ansible_set_config_file(
        msg='Ensure system crypto policy has OpenSSL library using TLS encryption',
        file='/etc/crypto-policies/back-ends/opensslcnf.config',
        parameter="MinProtocol",
        value='TLSv1.2',
        create='yes',
        separator="=",
        separator_regex="\s*=\s*",
        prefix_regex="^"
    )
}}}
```

ggbecker reviewed (changes requested):
Probably using macros is the way to go here. If we start using the same macro for other rules we can create a template for that. But in general, we were trying to move away from the `replace_or_append` function (even though there are still many occurrences in the project).

Additionally I'm getting inconsistent test results. It appears `update-crypto-policies` (in my RHEL8.3) is overwriting the contents of `/etc/crypto-policies/back-ends/opensslcnf.config` with some previous data. At a certain test, the command writes `#MinProtocol = TLSv1.2` back to the file, which was something related to a previous test. I'm not sure how we are suppose to populate these files and it might be a good idea to investigate that.

carlosmmatos commented on linux_os/guide/system/software/integrity/crypto/configure_openssl_tls_crypto_policy/rule.yml:
```diff
@@ -0,0 +1,34 @@
+    cce@rhel8: CCE-84255-9
+
+references:
+    nist: AC-17(2)
```
You're right.. the update-crypto-policies probably shouldn't run - this is something that should only be ran when setting a new crypto-policy, otherwise it will place all the default `/etc/crypto-policies/back-ends/...config` to their default state... I'll clean up to use the lineinfile template.

carlosmmatos commented on tests/data/profile_stability/rhel8/stig.profile:
```diff
@@ -1,6 +1,6 @@
 description: 'This profile contains configuration checks that align to the
 
-    DISA STIG for Red Hat Enterprise Linux 8.
+    DISA STIG for Red Hat Enterprise Linux 8 V1R1.
```
I'll update once I make my new changes and rebuild

carlosmmatos commented on linux_os/guide/system/software/integrity/crypto/configure_openssl_tls_crypto_policy/ansible/shared.yml:
```diff
@@ -0,0 +1,16 @@
+    insertafter: EOF
+
+- name: "Update system crypto policy for changes to take effect"
+  command:
+    cmd: "update-crypto-policies"
```
this was copy and paste.. but this is not needed anymore

carlosmmatos commented:
@ggbecker Interesting.. I was hoping that the `lineinfile` template would get the job done, but it clearly does not as it doesn't deal with duplicate keys in a config file.. I may just create a template on my next PR for what we seem to be wrapping around the `set_config_file` macro - which ensures the (at least on ansible side) - we call the ONLY lineinfile macro which is what we ultimately want.

Either way, I've implemented the macro suggestions and updated the ref and test profiles.

carlosmmatos commented:
/retest

ggbecker reviewed (approved):
LGTM

Both Bash and Ansible passed all test scenarios:

```
INFO - xccdf_org.ssgproject.content_rule_configure_openssl_tls_crypto_policy
INFO - Script correct_commented.fail.sh using profile (all) OK
INFO - Script empty_policy.fail.sh using profile (all) OK
INFO - Script incorrect_followed_by_correct.pass.sh using profile (all) OK
INFO - Script missing_file.fail.sh using profile (all) OK
INFO - Script correct_followed_by_incorrect.fail.sh using profile (all) OK
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script incorrect_policy.fail.sh using profile (all) OK
```
